From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 14 Sep 2014 20:08:27 +0100
Subject: arm64: compat: wire up memfd_create syscall for aarch32
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/760702

Implemented upstream by a97a42c47608d0bb6f2dfc2e162cc84a27beb43a,
but the arm64 compat layer looks rather different in 3.16.
---
--- a/arch/arm64/include/asm/unistd32.h
+++ b/arch/arm64/include/asm/unistd32.h
@@ -404,8 +404,11 @@ __SYSCALL(379, sys_finit_module)
 __SYSCALL(380, sys_sched_setattr)
 __SYSCALL(381, sys_sched_getattr)
 __SYSCALL(382, sys_renameat2)
+__SYSCALL(383, sys_ni_syscall) 		/* 383 for seccomp */
+__SYSCALL(384, sys_ni_syscall)			/* 384 for getrandom */
+__SYSCALL(385, sys_memfd_create)
 
-#define __NR_compat_syscalls		383
+#define __NR_compat_syscalls		386
 
 /*
  * Compat syscall numbers used by the AArch64 kernel.
